pages/mercury_tour_page.py

```diff
@@ -1,73 +1,126 @@
+class MainWelcomePage(BasePage):
 
-    def flight_det_sub(self):
+    def login_submit_verify(self, username, password):
+        """
+        Submit username and password in fields and submit
+        Returned next page
+        """
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.NAME, 'userName'))
+        ).send_keys(username)
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.NAME, 'password'))
+        ).send_keys(password)
+        self.driver.find_element_by_name('submit').click()
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR, 'img[src="images/banner2.gif"]'))
+        )
+
+    def open_flight_vertical_by_clicking(self):
+        """
+        Move to flights vertical
+        """
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.LINK_TEXT, 'Flights'))
+        ).click()
+
+    def select_passenger(self, pass_counts):
+        """
+        Enter any passcount in passenger field
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'passCount'))
+        )).select_by_visible_text(pass_counts)
+
+    def select_departing_location(self, from_port):
+        """
+        Enter from_port from field
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'fromPort'))
+        )).select_by_visible_text(from_port)
+
+    def select_departing_month(self, from_month):
+        """
+        Select from_month from dropdown
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'fromMonth'))
+        )).select_by_visible_text(from_month)
+
+    def select_departing_day(self, from_day):
+        """
+        Select from_day from dropdown
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'fromDay'))
+        )).select_by_visible_text(from_day)
+
+    def select_arriving_port(self, to_port):
+        """
+        Select to_port
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'toPort'))
+        )).select_by_visible_text(to_port)
+
+    def select_returning_month(self, to_month):
+        """
+        Select to_month from list
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'toMonth'))
+        )).select_by_visible_text(to_month)
+
+    def select_returning_day(self, to_day):
+        """
+        Select to_day from list
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'toDay'))
+        )).select_by_visible_text(to_day)
+
+    def select_service_class(self):
```

Comment:
either pass the service class in param or rename the method to `select_business_class`.

Reply:
I have already tried by taking service_class as a param and pass different options as 'busi, economy' etc but it wont work .As return value always shows "select_service_class()->None" .There is some website issue as its been discussed with raees. So right now i am just refactoring method